dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,321 @@
+                                          f' Glacier restore.')
+
+    def restore_portal_from_glacier(self, atid: str, days: int = 7) -> Union[dict, None]:
+        """ Resolves the given atid and restores it from glacier, returning the response if successful
+
+        :param atid: resource path to extract bucket, key information from
+        :param days: number of days to store in the temporary location
+        :return: response if successful or None
+        """
+        bucket, key = self.resolve_bucket_key_from_portal(atid)
+        return self.restore_s3_from_glacier(bucket, key, days=days)
+
+    def restore_s3_from_glacier(self, bucket: str, key: str, days: int = 7) -> Union[dict, None]:
+        """ Restores a file from glacier given the bucket, key and duration of restore
+
+        :param bucket: bucket where the file is stored
+        :param key: key under which the file is stored
+        :param days: number of days to store in the temporary location
+        :return: response if successful or None
+        """
+        try:
+            response = self.s3.restore_object(
+                Bucket=bucket,
+                Key=key,
+                RestoreRequest={'Days': days}
+            )
+            PRINT(f'Object {bucket}/{key} restored from Glacier storage class and will be available in S3'
+                  f' for {days} days')
+            return response
+        except Exception as e:
+            PRINT(f'Error restoring object {key} from Glacier storage class: {str(e)}')
+            return None
+
+    def extract_temporary_s3_location_from_restore_response(self, bucket: str, key: str) -> Union[dict, None]:
+        """ Extracts the S3 location that the restored file will be sent to given the bucket and key
+
+        :param bucket: bucket of original file location
+        :param key: key of original file location
+        :return: Restore object from s3
+        """
+        try:  # extract temporary location by heading object
+            response = self.s3.head_object(Bucket=bucket, Key=key)
+            restore = response.get('Restore')
+            if restore is None:
+                PRINT(f'Object {bucket}/{key} is not currently being restored from Glacier')
+                return None
+            if 'ongoing-request="false"' not in restore:
+                PRINT(f'Object {bucket}/{key} is still being restored from Glacier')
+                return None
+            restore_path = response.get('RestoreOutputPath')
+            if restore_path is None:
+                PRINT(f'Error: Could not determine the temporary location of restored object {bucket}/{key}')
+                return None
+            return restore
+        except Exception as e:
+            PRINT(f'Error copying object {bucket}/{key} back to its original location in S3: {str(e)}')
+            return None
+
+    def patch_file_lifecycle_status_to_standard(self, atid: str, status: str,
```

Comment:
Consider making `status` have a default of `uploaded` here, since that's correct for the vast majority of situations.